Set PYTHON_VERSION for buildkite jobs

twunderlich-grapl opened the pull request:
<!-- Thank you for your contribution to Grapl! Please take some time
to fill out this short template to help us review and understand your
PR. -->

### Which issue does this PR correspond to?
Fixes https://github.com/grapl-security/issue-tracker/issues/980
Fixes error in the CI pipeline: https://buildkite.com/grapl/grapl-merge/builds/760#0181f39e-3825-4e19-a267-0a28be427a4a
<!-- Please provide a link to the GitHub Issue this pull request is
meant to address. -->

### What changes does this PR make to Grapl? Why?
This fixes an error in https://buildkite.com/grapl/grapl-merge/builds/760#0181f39e-3825-4e19-a267-0a28be427a4a
Hardcode the python version in the buildkite pipeline. Previously we only defined PYTHON_VERSION in the makefile, but we have at least one job that doesn't go through Make and therefore needs the env variable defined twice
Long-term it would be nice to be able to grab this dynamically
<!-- Please describe the functional impact of these changes and the
rationale behind them. This will help us understand your approach to
solving the problem. -->

### How were these changes tested?
Manual buildkite run
<!-- Please describe how these changes were tested. There should be
sufficient detail that reviewers can replicate your testing
procedure. If the procedure is a manual one that's OK, your
description will help us work with you to get it automated. -->

Commit:
Set PYTHON_VERSION in buildkite and extract into lib

We need to set the PYTHON_VERSION in the Buildkite scripts
Instead of hardcoding the Python and Rust versions, I've extracted
the logic into a small lib for reuse

wimax-grapl commented on .buildkite/pipeline.verify.yml:
```diff
@@ -5,6 +5,7 @@ env:
   # TODO: Figure out a way for this to be sourced from our
   # rust-toolchain file
   RUST_VERSION: 1.62.0
+  PYTHON_VERSION: 3.10.5
```
if it's gonna be duplicated from .python-version, would be nice to have a 
`# also update .python-version and the same in .python-version`

christophermaier commented on .buildkite/scripts/lib/lang_version.sh:
```diff
@@ -0,0 +1,13 @@
+# This at least automatically keeps things in sync with with our Rust
+# toolchain.
+rust_version() {
+    grep channel src/rust/rust-toolchain.toml | sed -E 's/channel = "(.*)"/\1/g'
```
Let's not duplicate things: https://github.com/grapl-security/grapl/blob/main/build-support/rust_version.sh

twunderlich-grapl replied:
AH I didn't realize we had that. Updated

Commit:
Use existing script in build-support

There's an existing script in build-support to get the rust version,
we should be using that instead of rewriting the functionality again.
As such, I've also moved the Python logic there as well

christophermaier reviewed (approved):
I think Cue will eventually make all this madness go away soon, but for now, this is still a vast improvement over the status quo.

Commit:
Merge branch 'main' into twunderlich/fix-pipeline-python-version
